Source/DafnyRuntime/DafnyRuntimeJava/src/test/java/dafny/SequenceTest.java

```diff
@@ -22,16 +22,16 @@
   int[] testSequenceTakeArr = new int[]{1, 3, 2, 4, 2};
   int[] testSequenceDropArr = new int[]{4, 6, 5, 4, 1, 7};
   int[] testSequenceEmptyArr = new int[]{};
-  DafnySequence<Integer> testSequence = DafnySequence.of(testSequenceArr);
-  DafnySequence<Integer> testSequencePre = DafnySequence.of(testSequencePreArr);
-  DafnySequence<Integer> testSequenceNPre = DafnySequence.of(testSequenceNPreArr);
-  DafnySequence<Integer> testSequenceNPre2 = DafnySequence.of(testSequenceNPre2Arr);
-  DafnySequence<Integer> testSequenceSub = DafnySequence.of(testSequenceSubArr);
-  DafnySequence<Integer> testSequenceDrop = DafnySequence.of(testSequenceDropArr);
-  DafnySequence<Integer> testSequenceTake = DafnySequence.of(testSequenceTakeArr);
-  DafnySequence<Integer> testSequenceEmpty = DafnySequence.of(testSequenceEmptyArr);
-  DafnySequence<Integer> testCopy = DafnySequence.of(testSequenceArr);
-  DafnySequence<Integer> testWrappedSequence = DafnySequence.unsafeWrapRawArray(Type.INT, testSequenceArr);
+  DafnySequence<? extends Integer> testSequence = DafnySequence.of(testSequenceArr);
+  DafnySequence<? extends Integer> testSequencePre = DafnySequence.of(testSequencePreArr);
+  DafnySequence<? extends Integer> testSequenceNPre = DafnySequence.of(testSequenceNPreArr);
+  DafnySequence<? extends Integer> testSequenceNPre2 = DafnySequence.of(testSequenceNPre2Arr);
+  DafnySequence<? extends Integer> testSequenceSub = DafnySequence.of(testSequenceSubArr);
+  DafnySequence<? extends Integer> testSequenceDrop = DafnySequence.of(testSequenceDropArr);
+  DafnySequence<? extends Integer> testSequenceTake = DafnySequence.of(testSequenceTakeArr);
+  DafnySequence<? extends Integer> testSequenceEmpty = DafnySequence.of(testSequenceEmptyArr);
+  DafnySequence<? extends Integer> testCopy = DafnySequence.of(testSequenceArr);
+  DafnySequence<? extends Integer> testWrappedSequence = DafnySequence.unsafeWrapRawArray(Type.INT, testSequenceArr);
```

Comment:
I'm not sure it matters, but as Integer is final in Java, nothing (except Integer) can extend it.

Reply:
Apparently, the Java type system doesn't automatically treat `X<? extends Integer>` and `X<Integer>` as the same. Thus, it seems safer to always stick with the recipe of including `? extends` for co-variant type parameters.